proxy/src/main/java/com/velocitypowered/proxy/VelocityServer.java

```diff
@@ -118,6 +118,14 @@ public void start() {
             servers.register(new ServerInfo(entry.getKey(), AddressUtil.parseAddress(entry.getValue())));
         }
 
+        for (Map.Entry<String, List<String>> entry : configuration.getForcedHosts().entrySet()) {
+            for (String server : entry.getValue()) {
+                if (!servers.getServer(server).isPresent()) {
+                    logger.warn("Server '{}' for forced host '{}' does not exist", server, entry.getKey());
```

Comment:
IIRC invalid server entries cause premature disconnection from proxy, e.g if 3rd server is invalid and there are 7 servers in list, then player would get disconnected on 3rd server connection attempt.

So should list be copied and rewritten here or should logic be changed where disconnection happens?

Reply:
Your concern is unfounded. Velocity will try to connect to all seven servers, and only after all attempts are exhausted will it disconnect the server.

I would prefer this were a fatal error that would prevent the configuration from being loaded.